### ISSUE TITLE
Truth value ambigious. Use a.any() or a.all()

### ISSUE THREAD

**Issue by SELTENEERDEN** [LCA-ActivityBrowser/activity-browser#1267]
Hello, this problem just appeared again. I can not export my databases as well. When I click on "finish" in the export windows nothing happens. I would like to reinstall bw and ab. Will I loose my projects (databases), if I do not export them? Any other way to backup them? Can I just copy&paste files from a directory on my windows machine? If so, where is this folder?

Here is the error message by the way:

```
(base) C:\Windows\system32>activate ab

(ab) C:\Windows\system32>activity-browser
-hiding tab:History

DevTools listening on ws://127.0.0.1:3999/devtools/browser/9b99b1e6-fbb5-408d-bbe0-293d39788e38
-hiding tab:Activity Details
-hiding tab:Characterization Factors
-hiding tab:Graph Explorer
-hiding tab:LCA results
The Activity Browser log file can be found at C:\Users\benif\AppData\Local\ActivityBrowser\ActivityBrowser\Logs\ab_logs-2024-4-3_12-46-17.log
Activity Browser version: 2.9.7
Loading user settings:
dirpath already loaded
Reset project settings directory to:C:\Users\benif\AppData\Local\pylca\Brightway3\test.098f6bcd4621d373cade4e832627b4f6
Loaded project:test
Brightway2 data directory: C:\Users\benif\AppData\Local\pylca\Brightway3
Brightway2 active project: test
Qt Version:5.15.8
Remote debugging server started successfully. Try pointing a Chromium-based browser to http://127.0.0.1:3999
Traceback (most recent call last):
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\wizards\db_export_wizard.py", line 38, in accept
    self.perform_export()
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\wizards\db_export_wizard.py", line 52, in perform_export
    EXPORTERS[export_as](db_name, out_path)
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\bwutils\exporters.py", line 132, in store_database_as_package
    db.process()
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\bw2data\backends\peewee\database.py", line 445, in process
    if np.isnan(data['amount']) or np.isinf(data['amount']):
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()>
```
The environment data is
```
# packages in environment at C:\ProgramData\miniconda3\envs\ab:
#
# Name                    Version                   Build  Channel
activity-browser          2.9.7              pyhd8ed1ab_0    conda-forge
appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
arrow                     1.3.0              pyhd8ed1ab_0    conda-forge
asteval                   0.9.31             pyhd8ed1ab_0    conda-forge
astunparse                1.6.3              pyhd8ed1ab_0    conda-forge
blinker                   1.7.0              pyhd8ed1ab_0    conda-forge
brightway2                2.4.4              pyhd8ed1ab_0    conda-forge
brotli                    1.1.0                hcfcfb64_1    conda-forge
brotli-bin                1.1.0                hcfcfb64_1    conda-forge
brotli-python             1.1.0           py311h12c1d0e_1    conda-forge
brotlicffi                1.1.0.0         py311h12c1d0e_1    conda-forge
bw2analyzer               0.10               pyhd8ed1ab_0    conda-forge
bw2calc                   1.8.2           py311h1ea47a8_0    conda-forge
bw2data                   3.6.6              pyhd8ed1ab_0    conda-forge
bw2io                     0.8.10             pyhd8ed1ab_0    conda-forge
bw2parameters             1.1.0              pyhd8ed1ab_0    conda-forge
bw_migrations             0.2                pyhd8ed1ab_0    conda-forge
bzip2                     1.0.8                hcfcfb64_5    conda-forge
ca-certificates           2024.2.2             h56e8100_0    conda-forge
certifi                   2024.2.2           pyhd8ed1ab_0    conda-forge
cffi                      1.16.0          py311ha68e1ae_0    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
click                     8.1.7           win_pyh7428d3b_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
contourpy                 1.2.0           py311h005e61a_0    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
dill                      0.3.7              pyhd8ed1ab_0    conda-forge
docopt                    0.6.2                      py_1    conda-forge
eidl                      2.0.1              pyhd8ed1ab_0    conda-forge
eight                     1.0.1              pyhd8ed1ab_4    conda-forge
et_xmlfile                1.1.0              pyhd8ed1ab_0    conda-forge
fasteners                 0.17.3             pyhd8ed1ab_0    conda-forge
flask                     3.0.0              pyhd8ed1ab_0    conda-forge
fonttools                 4.47.2          py311ha68e1ae_0    conda-forge
freetype                  2.12.1               hdaf720e_2    conda-forge
future                    0.18.3             pyhd8ed1ab_0    conda-forge
gettext                   0.21.1               h5728263_0    conda-forge
glib                      2.78.3               h12be248_0    conda-forge
glib-tools                2.78.3               h12be248_0    conda-forge
gst-plugins-base          1.22.8               h001b923_1    conda-forge
gstreamer                 1.22.8               hb4038d2_1    conda-forge
icu                       73.2                 h63175ca_0    conda-forge
idna                      3.6                pyhd8ed1ab_0    conda-forge
importlib-metadata        7.0.1              pyha770c72_0    conda-forge
importlib_metadata        7.0.1                hd8ed1ab_0    conda-forge
inflate64                 1.0.0           py311ha68e1ae_0    conda-forge
intel-openmp              2023.2.0         h57928b3_50497    conda-forge
itsdangerous              2.1.2              pyhd8ed1ab_0    conda-forge
jinja2                    3.1.3              pyhd8ed1ab_0    conda-forge
kiwisolver                1.4.5           py311h005e61a_1    conda-forge
krb5                      1.21.2               heb0366b_0    conda-forge
lcms2                     2.16                 h67d730c_0    conda-forge
lerc                      4.0.0                h63175ca_0    conda-forge
libblas                   3.9.0              20_win64_mkl    conda-forge
libbrotlicommon           1.1.0                hcfcfb64_1    conda-forge
libbrotlidec              1.1.0                hcfcfb64_1    conda-forge
libbrotlienc              1.1.0                hcfcfb64_1    conda-forge
libcblas                  3.9.0              20_win64_mkl    conda-forge
libclang                  15.0.7          default_h77d9078_3    conda-forge
libclang13                15.0.7          default_h77d9078_3    conda-forge
libdeflate                1.19                 hcfcfb64_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libglib                   2.78.3               h16e383f_0    conda-forge
libhwloc                  2.9.1                h51c2c0f_0    conda-forge
libiconv                  1.17                 hcfcfb64_2    conda-forge
libjpeg-turbo             3.0.0                hcfcfb64_1    conda-forge
liblapack                 3.9.0              20_win64_mkl    conda-forge
libogg                    1.3.4                h8ffe710_1    conda-forge
libpng                    1.6.39               h19919ed_0    conda-forge
libsqlite                 3.44.2               hcfcfb64_0    conda-forge
libtiff                   4.6.0                h6e2ebb7_2    conda-forge
libvorbis                 1.3.7                h0e60522_0    conda-forge
libwebp                   1.3.2                hcfcfb64_1    conda-forge
libwebp-base              1.3.2                hcfcfb64_0    conda-forge
libxcb                    1.15                 hcd874cb_0    conda-forge
libxml2                   2.10.4               hc3477c8_0    conda-forge
libxslt                   1.1.37               h0192164_0    conda-forge
libzlib                   1.2.13               hcfcfb64_5    conda-forge
lxml                      4.9.2           py311h5942461_0    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markupsafe                2.1.3           py311ha68e1ae_1    conda-forge
matplotlib-base           3.8.2           py311h6e989c2_0    conda-forge
mkl                       2023.2.0         h6a75c08_50497    conda-forge
mrio_common_metadata      0.2.1              pyhd8ed1ab_0    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
multiprocess              0.70.15         py311ha68e1ae_1    conda-forge
multivolumefile           0.2.3              pyhd8ed1ab_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
networkx                  3.2.1              pyhd8ed1ab_0    conda-forge
numpy                     1.23.5          py311h95d790f_0    conda-forge
openjpeg                  2.5.0                h3d672ee_3    conda-forge
openpyxl                  3.1.2           py311ha68e1ae_1    conda-forge
openssl                   3.2.1                hcfcfb64_1    conda-forge
packaging                 23.2               pyhd8ed1ab_0    conda-forge
pandas                    2.1.4           py311hf63dbb6_0    conda-forge
pathos                    0.3.1              pyhd8ed1ab_0    conda-forge
patsy                     0.5.6              pyhd8ed1ab_0    conda-forge
pcre2                     10.42                h17e33f8_0    conda-forge
peewee                    3.17.0          py311had142b2_0    conda-forge
pillow                    10.2.0          py311h4dd8a23_0    conda-forge
pint                      0.21               pyhd8ed1ab_0    conda-forge
pip                       23.3.2             pyhd8ed1ab_0    conda-forge
pox                       0.3.3              pyhd8ed1ab_0    conda-forge
ppft                      1.7.6.7            pyhd8ed1ab_0    conda-forge
psutil                    5.9.7           py311ha68e1ae_0    conda-forge
pthread-stubs             0.4               hcd874cb_1001    conda-forge
pthreads-win32            2.9.1                hfa6e2cd_3    conda-forge
py7zr                     0.20.8             pyhd8ed1ab_1    conda-forge
pybcj                     1.0.2           py311ha68e1ae_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pycryptodomex             3.19.0          py311ha68e1ae_1    conda-forge
pypardiso                 0.4.3              pyhd8ed1ab_0    conda-forge
pyparsing                 3.1.1              pyhd8ed1ab_0    conda-forge
pyperclip                 1.8.2              pyhd8ed1ab_2    conda-forge
pyppmd                    1.1.0           py311h12c1d0e_0    conda-forge
pyprind                   2.11.2          py311h1ea47a8_1005    conda-forge
pyside2                   5.15.8          py311h2411a93_2    conda-forge
pysocks                   1.7.1              pyh0701188_6    conda-forge
python                    3.11.0          hcf16a7b_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-tzdata             2023.4             pyhd8ed1ab_0    conda-forge
python_abi                3.11                    4_cp311    conda-forge
pytz                      2023.3.post1       pyhd8ed1ab_0    conda-forge
pyxlsb                    1.0.10             pyhd8ed1ab_0    conda-forge
pyzstd                    0.15.9          py311he5d195f_1    conda-forge
qt-main                   5.15.8              h9e85ed6_18    conda-forge
qt-webengine              5.15.8               h4bf5c4e_4    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
salib                     1.4.7              pyhd8ed1ab_0    conda-forge
scipy                     1.11.4          py311h0b4df5a_0    conda-forge
seaborn                   0.13.1               hd8ed1ab_0    conda-forge
seaborn-base              0.13.1             pyhd8ed1ab_0    conda-forge
setuptools                69.0.3             pyhd8ed1ab_0    conda-forge
setuptools-scm            8.0.4              pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
stats_arrays              0.6.6              pyhd8ed1ab_0    conda-forge
statsmodels               0.14.1          py311h59ca53f_0    conda-forge
tabulate                  0.9.0              pyhd8ed1ab_1    conda-forge
tbb                       2021.9.0             h91493d7_0    conda-forge
texttable                 1.7.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.13               h5226925_1    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
tqdm                      4.66.1             pyhd8ed1ab_0    conda-forge
types-python-dateutil     2.8.19.20240106    pyhd8ed1ab_0    conda-forge
typing-extensions         4.9.0                hd8ed1ab_0    conda-forge
typing_extensions         4.9.0              pyha770c72_0    conda-forge
tzdata                    2023d                h0c530f3_0    conda-forge
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
unicodecsv                0.14.1             pyhd8ed1ab_2    conda-forge
unidecode                 1.3.8              pyhd8ed1ab_0    conda-forge
urllib3                   2.1.0              pyhd8ed1ab_0    conda-forge
vc                        14.3                hcf57466_18    conda-forge
vc14_runtime              14.38.33130         h82b7239_18    conda-forge
voluptuous                0.14.1             pyhd8ed1ab_0    conda-forge
vs2015_runtime            14.38.33130         hcb4865c_18    conda-forge
werkzeug                  3.0.1              pyhd8ed1ab_0    conda-forge
wheel                     0.42.0             pyhd8ed1ab_0    conda-forge
whoosh                    2.7.4           py311h1ea47a8_8    conda-forge
win_inet_pton             1.1.0              pyhd8ed1ab_6    conda-forge
wrapt                     1.16.0          py311ha68e1ae_0    conda-forge
xlrd                      2.0.1              pyhd8ed1ab_3    conda-forge
xlsxwriter                3.1.9              pyhd8ed1ab_0    conda-forge
xorg-libxau               1.0.11               hcd874cb_0    conda-forge
xorg-libxdmcp             1.1.3                hcd874cb_0    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge
zipfile-deflate64         0.2.0           py311ha68e1ae_4    conda-forge
zipp                      3.17.0             pyhd8ed1ab_0    conda-forge
zstd                      1.5.5                h12be248_0    conda-forge
```

_Originally posted by @SELTENEERDEN in https://github.com/LCA-ActivityBrowser/activity-browser/issues/780#issuecomment-2034241807_

**Comment by SELTENEERDEN** [comment]
> Please open a new issue for cases like this (ideally with a reference to the old one). It's just coincidence if somebody looks at or notices comment in old closed issues.
> 
> You can have multiple installations of the AB side by side using different conda environments. You can try something like this:
> 
> ```
> conda deactivate
> conda create -n ab2 -c conda-forge --solver libmamba activity-browser
> conda activate ab2
> activity-browser
> ```

And how will I be able to import my old DBs into that new instance? Or will they be part of the new instance automatically?

**Comment by mrvisscher** [comment]
@SELTENEERDEN,

Projects are stored in a centralized way, managed by Brightway2. You can see the folder in your logs: `Brightway2 data directory: C:\Users\benif\AppData\Local\pylca\Brightway3` Reinstalling the Activity Browser will not impact them, and you'll be able to open them directly from your fresh install.

Because of this I also doubt whether reinstalling BW and AB will solve your issue, as there seems to be something wrong with your project directly. But trying won't hurt.

Kind regards,

Marin

**Comment by SELTENEERDEN** [comment]
Hello, it is me again with the same bug again.

```
-hiding tab:History

DevTools listening on ws://127.0.0.1:3999/devtools/browser/25495cd2-bec2-41be-98b9-ad85359abce0
-hiding tab:Activity Details
-hiding tab:Characterization Factors
-hiding tab:Graph Explorer
-hiding tab:LCA results
The Activity Browser log file can be found at C:\Users\benif\AppData\Local\ActivityBrowser\ActivityBrowser\Logs\ab_logs-2024-10-1_12-56-6.log
Activity Browser version: 2.9.7
Loading user settings:
dirpath already loaded
Reset project settings directory to:C:\Users\benif\AppData\Local\pylca\Brightway3\test.098f6bcd4621d373cade4e832627b4f6
Loaded project:test
Brightway2 data directory: C:\Users\benif\AppData\Local\pylca\Brightway3
Brightway2 active project: test
Qt Version:5.15.8
Remote debugging server started successfully. Try pointing a Chromium-based browser to http://127.0.0.1:3999
Reset project settings directory to:C:\Users\benif\AppData\Local\pylca\Brightway3\default.c21f969b5f03d33d43e04f8f136e7682
Reset metadata.
Loaded project:default
Current shape and databases in the MetaDataStore:(0, 0)set()
Adding:cutoff391
+showing tab:Activity Details
Traceback (most recent call last):
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\LCA_results_tab.py", line 62, in generate_setup
    new_tab = LCAResultsSubTab(data, self)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\LCA_results_tabs.py", line 113, in __init__
    self.mlca, self.contributions, self.mc = calculations.do_LCA_calculations(data)
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\bwutils\calculations.py", line 28, in do_LCA_calculations
    mlca = MLCA(cs_name)
           ^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\bwutils\multilca.py", line 146, in __init__
    self.lca = self._construct_lca()
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\bwutils\multilca.py", line 189, in _construct_lca
    return bw.LCA(demand=self.func_units_dict, method=self.methods[0])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\bw2calc\lca.py", line 79, in __init__
    clean_databases()
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\bw2calc\utils.py", line 102, in clean_databases
    databases.clean()
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\bw2data\meta.py", line 121, in clean
    return _clean()
           ^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\bw2data\project.py", line 358, in writable_project
    return wrapped(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\bw2data\meta.py", line 114, in _clean
    Database(x).process()
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\bw2data\backends\peewee\database.py", line 445, in process
    if np.isnan(data['amount']) or np.isinf(data['amount']):
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

My Conda list output looks like:
```
# packages in environment at C:\ProgramData\miniconda3:
#
# Name                    Version                   Build  Channel
appdirs                   1.4.4                    pypi_0    pypi
archspec                  0.2.3              pyhd8ed1ab_0    conda-forge
asteval                   0.9.32                   pypi_0    pypi
astunparse                1.6.3                    pypi_0    pypi
boltons                   23.0.0          py311haa95532_0
brightway25               1.0.6                    pypi_0    pypi
brotli-python             1.0.9           py311hd77b12b_7
bw-migrations             0.2                      pypi_0    pypi
bw-processing             0.9                      pypi_0    pypi
bw2analyzer               0.11.8                   pypi_0    pypi
bw2calc                   2.0.dev16                pypi_0    pypi
bw2data                   4.0.dev33                pypi_0    pypi
bw2io                     0.9.dev26                pypi_0    pypi
bw2parameters             1.1.0                    pypi_0    pypi
bzip2                     1.0.8                he774522_0
ca-certificates           2024.2.2             h56e8100_0    conda-forge
certifi                   2024.2.2           pyhd8ed1ab_0    conda-forge
cffi                      1.16.0          py311h2bbff1b_0
charset-normalizer        2.0.4              pyhd3eb1b0_0
colorama                  0.4.6           py311haa95532_0
conda                     24.4.0          py311h1ea47a8_0    conda-forge
conda-content-trust       0.2.0           py311haa95532_0
conda-libmamba-solver     23.12.0            pyhd3eb1b0_1
conda-package-handling    2.2.0           py311haa95532_0
conda-package-streaming   0.9.0           py311haa95532_0
console_shortcut_miniconda 0.1.1                haa95532_1
contourpy                 1.2.1                    pypi_0    pypi
cryptography              41.0.7          py311h89fc84f_0
cycler                    0.12.1                   pypi_0    pypi
distro                    1.8.0           py311haa95532_0
et-xmlfile                1.1.0                    pypi_0    pypi
fmt                       10.2.1               h181d51b_0    conda-forge
fonttools                 4.50.0                   pypi_0    pypi
fs                        2.4.16                   pypi_0    pypi
idna                      3.4             py311haa95532_0
jsonpatch                 1.32               pyhd3eb1b0_0
jsonpointer               2.1                pyhd3eb1b0_0
kiwisolver                1.4.5                    pypi_0    pypi
krb5                      1.21.2               heb0366b_0    conda-forge
libarchive                3.7.2                h313118b_1    conda-forge
libcurl                   8.7.1                hd5e4a3a_0    conda-forge
libffi                    3.4.4                hd77b12b_0
libiconv                  1.17                 hcfcfb64_2    conda-forge
libmamba                  1.5.8                h3f09ed1_0    conda-forge
libmambapy                1.5.8           py311h0317a69_0    conda-forge
libsolv                   0.7.29               h0ea2cb4_0    conda-forge
libssh2                   1.11.0               h7dfc565_0    conda-forge
libxml2                   2.12.6               hc3477c8_2    conda-forge
libzlib                   1.2.13               hcfcfb64_5    conda-forge
lxml                      5.2.1                    pypi_0    pypi
lz4-c                     1.9.4                h2bbff1b_0
lzo                       2.10              he774522_1000    conda-forge
mamba                     1.5.8           py311h8cb466b_0    conda-forge
matplotlib                3.8.3                    pypi_0    pypi
matrix-utils              0.4.1                    pypi_0    pypi
menuinst                  2.0.1           py311hd77b12b_1
mrio-common-metadata      0.2.1                    pypi_0    pypi
numpy                     1.26.4                   pypi_0    pypi
openpyxl                  3.1.2                    pypi_0    pypi
openssl                   3.3.0                hcfcfb64_0    conda-forge
packaging                 23.1            py311haa95532_0
pandas                    2.2.1                    pypi_0    pypi
peewee                    3.17.1                   pypi_0    pypi
pillow                    10.3.0                   pypi_0    pypi
pint                      0.23                     pypi_0    pypi
pip                       23.3.1          py311haa95532_0
platformdirs              3.10.0          py311haa95532_0
pluggy                    1.0.0           py311haa95532_1
powershell_shortcut_miniconda 0.0.1                haa95532_1
pybind11-abi              4                    hd3eb1b0_1
pycosat                   0.6.6           py311h2bbff1b_0
pycparser                 2.21               pyhd3eb1b0_0
pyopenssl                 23.2.0          py311haa95532_0
pyparsing                 3.1.2                    pypi_0    pypi
pysocks                   1.7.1           py311haa95532_0
python                    3.11.5               he1021f5_0
python-dateutil           2.9.0.post0              pypi_0    pypi
python_abi                3.11                    2_cp311    conda-forge
pytz                      2024.1                   pypi_0    pypi
pyxlsb                    1.0.10                   pypi_0    pypi
reproc                    14.2.4               hd77b12b_1
reproc-cpp                14.2.4               hd77b12b_1
requests                  2.31.0          py311haa95532_0
ruamel.yaml               0.17.21         py311h2bbff1b_0
scipy                     1.13.0                   pypi_0    pypi
setuptools                68.2.2          py311haa95532_0
six                       1.16.0                   pypi_0    pypi
sqlite                    3.41.2               h2bbff1b_0
stats-arrays              0.6.6                    pypi_0    pypi
tabulate                  0.9.0                    pypi_0    pypi
tk                        8.6.12               h2bbff1b_0
tqdm                      4.65.0          py311h746a85d_0
truststore                0.8.0           py311haa95532_0
typing-extensions         4.10.0                   pypi_0    pypi
tzdata                    2024.1                   pypi_0    pypi
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
unidecode                 1.3.8                    pypi_0    pypi
urllib3                   1.26.18         py311haa95532_0
vc                        14.2                 h21ff451_1
vc14_runtime              14.38.33130         h82b7239_18    conda-forge
voluptuous                0.14.2                   pypi_0    pypi
vs2015_runtime            14.38.33130         hcb4865c_18    conda-forge
wheel                     0.41.2          py311haa95532_0
whoosh                    2.7.4                    pypi_0    pypi
win_inet_pton             1.1.0           py311haa95532_0
wrapt                     1.16.0                   pypi_0    pypi
xlrd                      2.0.1                    pypi_0    pypi
xlsxwriter                3.2.0                    pypi_0    pypi
xz                        5.4.5                h8cc25b3_0
yaml-cpp                  0.8.0                hd77b12b_0
zlib                      1.2.13               hcfcfb64_5    conda-forge
zstandard                 0.19.0          py311h2bbff1b_0
zstd                      1.5.5                hd43e919_0
```
This time unfortunately I can not so easily create a new project, import ecoinvent again and manually recreate all my projects, as they are a bit more complex this time. I would really need to find a solution to fix this. I am kind of at the finish line of my PhD-Thesis and panicking a bit because this bug turned up again. I also find it quite suprising, that I am the only one who constantly runs into this bug. I do not think, that anything about my windows computer and miniconda is so different from anyone elses xD.

**Comment by mrvisscher** [comment]
I think you've sent the wrong environment, this one doesn't contain the Activity Browser. 

Have you been using Brightway25 and Brightway2 in parallel? Because that's not supported by the AB right now.

Kind regards,

Marin

**Comment by SELTENEERDEN** [comment]
Hi Marin,
thanks for your quick reply, as always. I really appreciate the effort you're putting in.
I updated the conda list output with "ab" activated, and it shows that I have Brightway 2.4.4 installed.

I think I found something else that might be causing the bug. It occurred after I duplicated one of my processes. I had already noticed something strange after duplicating and renaming the process. When I entered a formula in the biosphere flows, it didn't affect the amount column. After deleting this process, the project works again. Interestingly, before I deleted the process, the bug impacted all calculations, even those without the specific process. It also crashed when I used any random ecoinvent process.

Kind regards,

Benni

**Comment by mrvisscher** [comment]
Okay, yeah, parameters. 

So the reason the amount didn't update is probably because there was no activity-parameter in place to hook the parameterized exchange into (please don't ask me why parameters work like this). We've changed the AB's approach to this in >2.10, so I'm not entirely sure whether your issue still exists in newer versions. I'm also not sure whether you should risk updating in that regard (though you could always make a project backup and roll-back ofcourse).

The bug probably affected all your calculations because databases are processed as a whole, not on a per activity basis. If one activity fails to process, the whole database fails to process.

Do I understand correctly that you've been able to solve the problem in it's entirety for your project?

**Comment by SELTENEERDEN** [comment]
Yes, you understood correctly. I don’t dare to update right now :)

Last time the bug happened, I created a new project and started from scratch. Just now, I opened the old, corrupted project to check if there were any activities where I couldn't open the activity details (by double-clicking on them). And voilà, I found two such activities. After deleting them, the old project now also works. I don't remember if the bug occurred after duplicating a process back then, but I wouldn’t rule it out. Maybe the issue is related to duplicating processes.

**Comment by mrvisscher** [comment]
Does it show the same error when you try to open the activity details in these processes?

**Comment by SELTENEERDEN** [comment]
To me the error looks a bit different, but I am not an expert. I do not even know, if this is the correct part of the code that I copy&pasted.

```
Loaded project:test
Current shape and databases in the MetaDataStore:(0, 0)set()
Adding:Allokation verstehen
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
Current shape and databases in the MetaDataStore:(3, 8){'Allokation verstehen'}
Adding:co2-malaga
+showing tab:Activity Details
+showing tab:Activity Details
Current shape and databases in the MetaDataStore:(5, 8){'Allokation verstehen', 'co2-malaga'}
Adding:kevitsa
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
-hiding tab:Activity Details
Current shape and databases in the MetaDataStore:(16, 8){'Allokation verstehen', 'co2-malaga', 'kevitsa'}
Adding:phu-kam, cu-ag-au-mine
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
Current shape and databases in the MetaDataStore:(23, 9){'Allokation verstehen', 'co2-malaga', 'phu-kam, cu-ag-au-mine', 'kevitsa'}
Adding:rönnskär
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
+showing tab:Activity Details
-hiding tab:Activity Details
Current shape and databases in the MetaDataStore:(33, 9){'co2-malaga', 'phu-kam, cu-ag-au-mine', 'Allokation verstehen', 'rönnskär', 'kevitsa'}
Adding:rönnskär ecoinvent
Traceback (most recent call last):
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 64, in safe_open_activity_tab
    self.open_activity_tab(key)
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 41, in open_activity_tab
    new_tab = ActivityTab(key, read_only=read_only)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 181, in __init__
    self.populate()
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 216, in populate
    self.production.model.sync(self.activity.production())
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 46, in sync
    self._dataframe = pd.DataFrame([
                                   ^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 46, in <listcomp>
    self._dataframe = pd.DataFrame([
                                   ^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 45, in <genexpr>
    data = (self.create_row(exc) for exc in self.exchanges)
            ^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 218, in create_row
    row = super().create_row(exchange)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 61, in create_row
    "Amount": float(exchange.get("amount", 1)),
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: float() argument must be a string or a real number, not 'tuple'

Traceback (most recent call last):
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 64, in safe_open_activity_tab
    self.open_activity_tab(key)
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 41, in open_activity_tab
    new_tab = ActivityTab(key, read_only=read_only)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 181, in __init__
    self.populate()
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 216, in populate
    self.production.model.sync(self.activity.production())
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 46, in sync
    self._dataframe = pd.DataFrame([
                                   ^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 46, in <listcomp>
    self._dataframe = pd.DataFrame([
                                   ^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 45, in <genexpr>
    data = (self.create_row(exc) for exc in self.exchanges)
            ^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 218, in create_row
    row = super().create_row(exchange)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 61, in create_row
    "Amount": float(exchange.get("amount", 1)),
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: float() argument must be a string or a real number, not 'tuple'

Traceback (most recent call last):
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 64, in safe_open_activity_tab
    self.open_activity_tab(key)
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 41, in open_activity_tab
    new_tab = ActivityTab(key, read_only=read_only)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 181, in __init__
    self.populate()
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\layouts\tabs\activity.py", line 217, in populate
    self.technosphere.model.sync(self.activity.technosphere())
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 46, in sync
    self._dataframe = pd.DataFrame([
                                   ^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 46, in <listcomp>
    self._dataframe = pd.DataFrame([
                                   ^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 45, in <genexpr>
    data = (self.create_row(exc) for exc in self.exchanges)
            ^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 241, in create_row
    row = super().create_row(exchange)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\miniconda3\envs\ab\Lib\site-packages\activity_browser\ui\tables\models\activity.py", line 61, in create_row
    "Amount": float(exchange.get("amount", 1)),
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: float() argument must be a string or a real number, not 'tuple'
```

**Comment by mrvisscher** [comment]
Different error, but same source. 

This is a really strange/intriguing issue. If you get the chance, can you send me a backup of a project where this issue occurs?

```
conda activate ab
python
import bw2io
bw2io.backup_project_directory("YOUR_PROJECT_NAME_HERE")
```

This should create a .tar.gz in your home folder which you can send to me at m.r.visscher@cml.leidenuniv.nl

**Comment by SELTENEERDEN** [comment]
Now that I deleted the activities that did not open up, I do not have any project where the issue still occurs. I am so sorry. If it turns up again, I will create such a backup!
